### PR TITLE
EEDA: Replace usage of deprecated paths/field names

### PR DESCRIPTION
**Pull request by saicheems**
## What does this PR do?
Replace:
- usage of `path` with `name`
- calls to "/v1/assets/**" with "/v1/projects/\*/assets/\*\*"
- calls to "/v1/assets:listImages" with "/v1/projects/\*/assets/\*\*:listImages"
- calls to "/v1/assets:getPixels" with "/v1/projects/\*/assets/\*\*:getPixels"

## Tasklist

 - [x] Add test case(s)
 - [x] Review
 - [x] Adjust for comments
 - [x] All CI builds and checks have passed

### CODE REVIEW

File: gdal/frmts/eeda/eeda.h [hunk 1]
```diff
@@ -85,6 +85,7 @@ class GDALEEDABaseDataset: public GDALDataset
             GIntBig     m_nExpirationTime;
 
             char      **GetBaseHTTPOptions();
+            CPLString   ConvertPathToName(const CPLString& path);
```
Comment (rouault) [inline gdal/frmts/eeda/eeda.h]: static

File: gdal/frmts/eeda/eedadataset.cpp [hunk 2]
```diff
@@ -187,6 +187,14 @@ GDALEEDALayer::GDALEEDALayer(GDALEEDADataset* poDS,
     m_poFeatureDefn->GetGeomFieldDefn(0)->SetSpatialRef(poSRS);
     poSRS->Release();
 
+    {
+        OGRFieldDefn oFieldDefn("name", OFTString);
+        m_poFeatureDefn->AddFieldDefn(&oFieldDefn);
+    }
+    {
+        OGRFieldDefn oFieldDefn("id", OFTString);
```
Comment (rouault) [inline gdal/frmts/eeda/eedadataset.cpp]: where is this field populated ?
If it is still needed and populated, then drv_eeda.html should mention it

File: gdal/frmts/eeda/eedaidataset.cpp [hunk 3]
```diff
@@ -1276,6 +1274,8 @@ bool GDALEEDAIDataset::Open(GDALOpenInfo* poOpenInfo)
         m_osAsset = papszTokens[1];
         CSLDestroy(papszTokens);
     }
+    m_osAsset = this->ConvertPathToName(m_osAsset);
```
Comment (rouault) [inline gdal/frmts/eeda/eedaidataset.cpp]: this-> useless

**Comment by rouault** [comment]
autotest/ogr/ogr_eeda.py:429: local variable 'ds' is assigned to but never used

Python3 CI config ( https://api.travis-ci.com/v3/job/162609668/log.txt ) crashed with a
```
./test_gdal_fuzzer
terminate called after throwing an instance of 'std::regex_error'
  what():  regex_error
make: *** [check] Aborted (core dumped)
```

Similarly trusty_32bit config ( https://travis-ci.com/OSGeo/gdal/jobs/162609673 ) crashed with
```
./gdal_unit_test
terminate called after throwing an instance of 'std::regex_error'
  what():  regex_error
make: *** [quick_test] Aborted (core dumped)
```

File: gdal/frmts/eeda/eedacommon.cpp [hunk 4]
```diff
@@ -29,10 +29,10 @@
 #include "cpl_http.h"
 #include "eeda.h"
 #include "ogrgeojsonreader.h"
+#include "regex.h"
```
[the remark this answers is not in the document]
Comment (saicheems) [reply gdal/frmts/eeda/eedacommon.cpp]: Thanks for pointing out the Windows failure.  I was hoping to get away with regex, but oh well, I to use `find` as you suggested.

I'll squash once I've confirmed that CI passed.

[the remark this answers is not in the document]
Comment (rouault) [reply gdal/frmts/eeda/eedacommon.cpp]: #include <regex.h> still there

[the remark this answers is not in the document]
Comment (saicheems) [reply gdal/frmts/eeda/eedacommon.cpp]: Thanks, removed.

File: gdal/frmts/eeda/eedadataset.cpp (hunk 2 above)
Comment (saicheems) [reply gdal/frmts/eeda/eedadataset.cpp]: Thanks, I misunderstood how this ended up in the output. I added code to populate them and included descriptions in drv_eeda.html.

**Pull request merged**